CodeGeneration/Sources/SyntaxSupport/AvailabilityNodes.swift

```diff
@@ -118,42 +118,51 @@ public let AVAILABILITY_NODES: [Node] = [
     ]
   ),
 
-  // version-tuple -> integer-literal
-  //                | float-literal
-  //                | float-literal '.' integer-literal
+  // version-tuple-element -> '.' interger-literal
   Node(
-    name: "VersionTuple",
-    nameForDiagnostics: "version tuple",
-    description: "A version number of the form major.minor.patch in which the minor and patch part may be omitted.",
+    name: "VersionComponent",
+    nameForDiagnostics: "version tuple element",
+    description: "An element to represent a dot and number pair",
     kind: "Syntax",
     children: [
       Child(
-        name: "Major",
-        kind: .token(choices: [.token(tokenKind: "IntegerLiteralToken")]),
-        description: "The major version."
-      ),
-      Child(
-        name: "MinorPeriod",
+        name: "Period",
         kind: .token(choices: [.token(tokenKind: "PeriodToken")]),
-        description: "If the version contains a minor number, the period separating the major from the minor number.",
-        isOptional: true
+        description: "The period of this pair"
       ),
       Child(
-        name: "Minor",
+        name: "Number",
         kind: .token(choices: [.token(tokenKind: "IntegerLiteralToken")]),
-        description: "The minor version if specified.",
-        isOptional: true
+        description: "The number of this pair"
       ),
+    ]
+  ),
+
+  // version-list -> version-tuple-element version-list?
+  Node(
+    name: "VersionComponentList",
+    nameForDiagnostics: "version list",
```

Comment:
“version list” is a big misleading here IMO because it’s not a list of versions, it’s just the components of a single version. I would just not provide a `nameForDiagnositcs` and then the name for diagnostics will default to the parent, which is `version tuple`